### PR TITLE
fix(kad): re-expose Distance as KBucketDistance

### LINKED ISSUE #4108: make kad::kbucket::Distance public again?
We rely on the public `kbucket::Key` and use the `.distance()` in some functionality. 

https://github.com/maidsafe/safe_network/blob/main/sn_protocol/src/lib.rs#L113C1-L116C6

I was have a look at upgrading but as of 0.52 `Distance` is no longer exposed. #4103 

Is there a blocker to reenabling that as a `pub` API?

### PR DESCRIPTION
**Pull request by joshuef**
## Description
Adds back in Distance as a public type. Follows `KBucketKey` naming for clarity.

Resolves #4108.

## Change checklist

- [x] I have performed a self-review of my own code
- [-] I have made corresponding changes to the documentation
- [-] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: protocols/kad/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
-## 0.44.0 
+## 0.44.1
```
Comment (thomaseizinger) [inline protocols/kad/CHANGELOG.md:1]:
```suggestion
## 0.44.1 - unreleased
```

File: protocols/kad/CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,8 @@
...
+- Expose KBucketDistance again
```
Comment (thomaseizinger) [inline protocols/kad/CHANGELOG.md]:
```suggestion
- Expose KBucketDistance.
```

Missing PR reference.

**Review by thomaseizinger — APPROVED** [review]
Minor comments, otherwise LGTM

**Comment by joshuef** [comment]
Committed those the now. Thanks for the review and the hard work on libp2p in general! 💪

**Review by mxinden — APPROVED** [review]
Thank you!

**Comment by mxinden** [comment]
[Tagged](https://github.com/libp2p/rust-libp2p/releases/tag/libp2p-kad-v0.44.1) and [published](https://crates.io/crates/libp2p-kad/0.44.1).

**Pull request merged**